src/_pytest/main.py

```diff
@@ -387,9 +398,14 @@ def __init__(self, config):
         self._initialpaths = frozenset()
         # Keep track of any collected nodes in here, so we don't duplicate fixtures
         self._node_cache = {}
+        self._bestrelpathcache = _bestrelpath_cache(config.rootdir)
 
         self.config.pluginmanager.register(self, name="session")
 
+    def _node_location_to_relpath(self, node_path):
+        # bestrelpath is a quite slow function
+        return self._bestrelpathcache[node_path]
```

Comment:
What about using lru_cache here?
Or mark is as TODO, since py27 does not use it.
(we could also backport it for py27, of course, if we're starting to use it more/here)

Reply:
lru cache has a significantly higher complexity - for a full cache (like this one) its overcill ^^
that being said - in a few iterations the cache will be gone for more effective ways